### ISSUE TITLE
[Combobox] Information about options being 'selected' and 'not selected' are not announced by screen readers

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3256]
### 😯 Current Behavior

For MacOs Firefox and Safari:

https://github.com/user-attachments/assets/8c8d9639-8936-42c7-b5b4-f061bb411e0b

https://github.com/user-attachments/assets/6c92c24d-d5b3-4cb8-b992-fd4a1b0d7577

For Win11 Firefox:

https://github.com/user-attachments/assets/0899f781-427f-4513-a181-78feeaef0443

This issue can be reproduce for stories: 
1. Combobox Multiple Group Selected
2. Combobox Multiple Groups
3. Combobox Multiple Groups With Group Selected
4. Combobox Single Groups

### 🤔 Expected Behavior/possible solution?

https://github.com/user-attachments/assets/7555738c-9a8d-4481-ac28-5fda398e36cd

Behavior like that seen on Chrome is the expected one. Screen reader can announce that elements are not selected or that they are selected. It's sufficient if only one type of information is being announced like only not selected items are announced as such but all the selected ones are announced just as menu items/options without information that they are selected.

### 🖥️ Steps to Reproduce

This issue can be reproduce for stories: 
1. Combobox Multiple Group Selected
2. Combobox Multiple Groups
3. Combobox Multiple Groups With Group Selected
4. Combobox Single Groups

**Steps to follow:** 
1. With screen reader turned on using Safari or Firefox go to Combobox story: https://storybook.sanomalearning.design/?path=/story/form-combobox-multiple--group-selected
2. Expand list of options and check if screen reader announces 'not selected' with option name
3. Select any option and check if selected option is announced by screen reader as 'selected'

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/form-combobox-multiple--group-selected

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Other

### What browsers are you seeing the problem on?

Safari, Firefox

### What operating system are you using?

macOs and Win11

### Component version

v0.1.7

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by a11ymiko** [comment]
Safari seems to not add "selected=false" state in Accessibility tab for `aria-selected=false`. Since options in these variant of comboboxes right now only have `aria-selected=false` or no `aria-selected` at all (when selected) there is no difference for VoiceOver between selected and not selected options. 

<img width="568" height="331" alt="Image" src="https://github.com/user-attachments/assets/cc1a1e11-1127-4f02-9f7e-5ade78664fef" />

Adding `aria-selected=true` to selected options would fix that issue for Safari and will work well in other browsers.

**Comment by Diaan** [comment]
https://share.google/aimode/E8jGPY0fySG5H5RPK

This problem also occurs in the select component when the option is in a group.
